serializer_impl.hh

```diff
@@ -791,15 +791,56 @@ unknown_variant_type deserialize(Input& in, boost::type<unknown_variant_type>) {
 // using a range.
 // Use begin() and end() to iterate through the frozen vector,
 // deserializing (or skipping) one element at a time.
-template <typename T, typename InputStream = utils::input_stream>
+template <typename T, bool IsForward=true>
```

Comment:
I wonder how that would work for consuming e.g. mutation_partition_view in reverse.
Would one construct a reversed vector_deserializer from the forward one that we work with today?
Initially what I had in mind is constructing the substreams vector first time rbegin is called on the (bidirectional) vector_deserializer but maybe there's no need for that.

Reply:
We now construct a forward vector deserializer and we could just make a reverse vector deserializer. It seems that the vector deserializer is used straight away after instatiating, so there's no much difference between that and by constructing substreams when calling begin (not rbegin!). I wanted to contain it in the vector deserializer, as otherwise I would need to care about which particular iterator holds the substreams.